pkg/3scale/amp/component/backend_options.go

```diff
@@ -0,0 +1,98 @@
+package component
+
+import (
+	oprand "github.com/3scale/3scale-operator/pkg/crypto/rand"
+	"github.com/go-playground/validator/v10"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type BackendOptions struct {
+	ServiceEndpoint              string `validate:"required"`
+	RouteEndpoint                string `validate:"required"`
+	StorageURL                   string `validate:"required"`
+	QueuesURL                    string `validate:"required"`
+	StorageSentinelHosts         string
+	StorageSentinelRole          string
+	QueuesSentinelHosts          string
+	QueuesSentinelRole           string
+	ListenerResourceRequirements v1.ResourceRequirements `validate:"-"`
+	WorkerResourceRequirements   v1.ResourceRequirements `validate:"-"`
+	CronResourceRequirements     v1.ResourceRequirements `validate:"-"`
```

Comment:
Why are this fields not required? I guess this is related to the question I made about what's the general criteria of considering some option required/non-required and what does that means for us and the validator and where the responsabilities lie

Reply:
Yes, above comments my answer. 

Let me add that `validate:"-"` means skip validation and zero value is a valid value.